[BUG] test_parquet_read_encryption fails

**Describe the bug**
When I run the python integration tests on Spark 3.2.0 or spark_3.3.0

```
env -u SPARK_CONF_DIR SPARK_HOME=~/spark_3.2.0/ ./run_pyspark_from_build.sh -k 'parquet and encryption'
```

Most of them fail with an error like 

```
Caused by: java.lang.ClassNotFoundException: Class org.apache.parquet.crypto.keytools.mocks.InMemoryKMS not found
	at org.apache.hadoop.conf.Configuration.getClassByName(Configuration.java:2571)
	at org.apache.parquet.hadoop.util.ConfigurationUtil.getClassFromConfig(ConfigurationUtil.java:33)
```

All I did was to run buildall.sh and then cd into the integration tests directory and run the tests.

I looks like 

```
@pytest.mark.skipif(os.environ.get('INCLUDE_PARQUET_HADOOP_TEST_JAR', 'false') == 'false', reason='INCLUDE_PARQUET_HADOOP_TEST_JAR is disabled')
```

is the culprit.  It is being set to true in the run_pyspark_from_build.sh without ever even checking if the hadoop dependency was placed where it is expected to be. But honestly I have no idea because I don't know what jar is expected to be on the classpath. It could be the fact that I build all of the versions and somehow the wrong dependency was downloaded. Either way this is a bad user experience.

```
FAILED ../../src/main/python/parquet_test.py::test_parquet_read_encryption[reader_confs3-] - py4j.protocol.Py4JJavaError: An error occurred while calling o310.parquet.
FAILED ../../src/main/python/parquet_test.py::test_parquet_read_encryption[reader_confs0-] - py4j.protocol.Py4JJavaError: An error occurred while calling o310.parquet.
FAILED ../../src/main/python/parquet_test.py::test_parquet_read_encryption[reader_confs1-parquet] - py4j.protocol.Py4JJavaError: An error occurred while calling o310.parquet.
FAILED ../../src/main/python/parquet_test.py::test_parquet_read_encryption[reader_confs2-parquet] - py4j.protocol.Py4JJavaError: An error occurred while calling o310.parquet.
FAILED ../../src/main/python/parquet_test.py::test_parquet_read_encryption[reader_confs5-] - py4j.protocol.Py4JJavaError: An error occurred while calling o310.parquet.
FAILED ../../src/main/python/parquet_test.py::test_parquet_read_encryption[reader_confs2-] - py4j.protocol.Py4JJavaError: An error occurred while calling o310.parquet.
FAILED ../../src/main/python/parquet_test.py::test_parquet_read_encryption[reader_confs4-] - py4j.protocol.Py4JJavaError: An error occurred while calling o310.parquet.
FAILED ../../src/main/python/parquet_test.py::test_parquet_read_encryption[reader_confs3-parquet] - py4j.protocol.Py4JJavaError: An error occurred while calling o310.parquet.
FAILED ../../src/main/python/parquet_test.py::test_parquet_read_encryption[reader_confs0-parquet] - py4j.protocol.Py4JJavaError: An error occurred while calling o310.parquet.
FAILED ../../src/main/python/parquet_test.py::test_parquet_read_encryption[reader_confs1-] - py4j.protocol.Py4JJavaError: An error occurred while calling o310.parquet.
FAILED ../../src/main/python/parquet_test.py::test_parquet_read_encryption[reader_confs4-parquet] - py4j.protocol.Py4JJavaError: An error occurred while calling o310.parquet.
FAILED ../../src/main/python/parquet_test.py::test_parquet_read_encryption[reader_confs5-parquet] - py4j.protocol.Py4JJavaError: An error occurred while calling o400.parquet.
```

--- a/integration_tests/run_pyspark_from_build.sh
+++ b/integration_tests/run_pyspark_from_build.sh
@@ -57,7 +57,10 @@ else
     else
         AVRO_JARS=$(echo "$SCRIPTPATH"/target/dependency/spark-avro*.jar)
         PARQUET_HADOOP_TESTS=$(echo "$SCRIPTPATH"/target/dependency/parquet-hadoop*.jar)
-        export INCLUDE_PARQUET_HADOOP_TEST_JAR=true
+        REGEX="parquet-hadoop-1\.([0-9]{2})"
+        [[ $PARQUET_HADOOP_TESTS =~ $REGEX  ]]
+        # Make sure we have Parquet version >= 1.12 in the dependency
+        export INCLUDE_PARQUET_HADOOP_TEST_JAR=$([ "${BASH_REMATCH[1]}" -ge "12" ] && echo true || echo false)
         PLUGIN_JARS=$(echo "$SCRIPTPATH"/../dist/target/rapids-4-spark_*.jar)
         # the integration-test-spark3xx.jar, should not include the integration-test-spark3xxtest.jar
         TEST_JARS=$(echo "$SCRIPTPATH"/target/rapids-4-spark-integration-tests*-$INTEGRATION_TEST_VERSION.jar)

--- a/integration_tests/run_pyspark_from_build.sh
+++ b/integration_tests/run_pyspark_from_build.sh
@@ -57,10 +57,10 @@ else
     else
         AVRO_JARS=$(echo "$SCRIPTPATH"/target/dependency/spark-avro*.jar)
         PARQUET_HADOOP_TESTS=$(echo "$SCRIPTPATH"/target/dependency/parquet-hadoop*.jar)
-        REGEX="parquet-hadoop-1\.([0-9]{2})"
-        [[ $PARQUET_HADOOP_TESTS =~ $REGEX  ]]
+        MIN_PARQUET_JAR="$SCRIPTPATH/target/dependency/parquet-hadoop-1.12.0-tests.jar"
         # Make sure we have Parquet version >= 1.12 in the dependency
-        export INCLUDE_PARQUET_HADOOP_TEST_JAR=$([ "${BASH_REMATCH[1]}" -ge "12" ] && echo true || echo false)
+        LOWEST_PARQUET_JAR=$(echo -e "$MIN_PARQUET_JAR\n$PARQUET_HADOOP_TESTS" | sort -V | head -1)
+        export INCLUDE_PARQUET_HADOOP_TEST_JAR=$([[ "$LOWEST_PARQUET_JAR" == "MIN_PARQUET_JAR" ]] && echo true || echo false)
         PLUGIN_JARS=$(echo "$SCRIPTPATH"/../dist/target/rapids-4-spark_*.jar)
         # the integration-test-spark3xx.jar, should not include the integration-test-spark3xxtest.jar
         TEST_JARS=$(echo "$SCRIPTPATH"/target/rapids-4-spark-integration-tests*-$INTEGRATION_TEST_VERSION.jar)

